core/src/com/unciv/ui/screens/worldscreen/unit/actions/UnitActions.kt

```diff
@@ -73,22 +93,47 @@ object UnitActions {
         UnitActionType.SpreadReligion to UnitActionsReligion::getSpreadReligionActions,
         UnitActionType.RemoveHeresy to UnitActionsReligion::getRemoveHeresyActions,
         UnitActionType.TriggerUnique to UnitActionsFromUniques::getTriggerUniqueActions,
-        UnitActionType.AddInCapital to UnitActionsFromUniques::getAddInCapitalActions
+        UnitActionType.AddInCapital to UnitActionsFromUniques::getAddInCapitalActions,
+        UnitActionType.GiftUnit to UnitActions::getGiftActions
     )
 
-    private fun shouldAutomationBePrimaryAction(unit:MapUnit) = unit.cache.hasUniqueToBuildImprovements || unit.hasUnique(UniqueType.AutomationPrimaryAction)
+    /** Gets the preferred "page" to display a [UnitAction] of type [unitActionType] on, possibly dynamic depending on the state or situation [unit] is in. */
+    fun getActionDefaultPage(unit: MapUnit, unitActionType: UnitActionType) =
+        actionTypeToPageGetter[unitActionType]?.invoke(unit) ?: unitActionType.defaultPage
+
+    /** Only for action types that wish to change their "More/Back" page position depending on context.
+     *  All others get a defaultPage statically from [UnitActionType].
+     */
+    private val actionTypeToPageGetter = linkedMapOf<UnitActionType, (unit: MapUnit) -> Int>(
+        UnitActionType.Automate to { unit ->
+            if (unit.cache.hasUniqueToBuildImprovements || unit.hasUnique(UniqueType.AutomationPrimaryAction)) 0 else 1
+        },
+        UnitActionType.Fortify to { unit ->
+            // Fortify moves to second page if current action is FortifyUntilHealed or if unit is wounded and it's not already the current action
+            if (unit.isFortifyingUntilHealed() || unit.health < 100 && !(unit.isFortified() && !unit.isActionUntilHealed())) 1 else 0
+        },
+        UnitActionType.FortifyUntilHealed to { unit ->
+            // FortifyUntilHealed only moves to the second page if Fortify is the current action
+            if (unit.isFortified() && !unit.isActionUntilHealed()) 1 else 0
+        },
+        UnitActionType.Sleep to { unit ->
+            // Sleep moves to second page if current action is SleepUntilHealed or if unit is wounded and it's not already the current action
+            if (unit.isSleepingUntilHealed() || unit.health < 100 && !(unit.isSleeping() && !unit.isActionUntilHealed())) 1 else 0
+        },
+        UnitActionType.SleepUntilHealed to { unit ->
+            // SleepUntilHealed only moves to the second page if Sleep is the current action
+            if (unit.isSleeping() && !unit.isActionUntilHealed()) 1 else 0
+        },
+        UnitActionType.Explore to { unit ->
+            if (unit.isCivilian()) 1 else 0
+        },
```

Comment:
This really feels like something that should be defined where the default page is...?
Probably fine as is